File: amprobe/service/schema/account.go
// Package schema
// Date       : 2024/9/4 15:14
// Author     : Amu
// Description:
package schema

type User struct {
	ID        string `json:"id"`
	Username  string `json:"username"`
	Remark    string `json:"remark"`
	IsAdmin   int    `json:"isAdmin"`
	Status    int    `json:"status"`
	Roles     []Role `json:"roles"`
	CreatedAt string `json:"created_at"`
}

type Role struct {
	ID        string     `json:"id"`
	Name      string     `json:"name"`
	Remark    string     `json:"remark"`
	Status    int        `json:"status"`
	Resources []Resource `json:"resources"`
	CreatedAt string     `json:"created_at"`
}

type Resource struct {
	ID     string `json:"id"`
	Name   string `json:"name"`
	Path   string `json:"path"`
	Method string `json:"method"`
	Status int    `json:"status"`
}

type UserQueryArgs struct {
	ID        string   `json:"id,omitempty"`
	IDs       []string `json:"ids,omitempty"`
	Username  string   `json:"username,omitempty"`
	Usernames []string `json:"usernames,omitempty"`
	Status    int      `json:"status,omitempty"`
}

type UserQueryReply struct {
	Data  []User `json:"data"`
	Total int    `json:"total"`
}

type UserCreateArgs struct {
	Username string   `json:"username" validate:"required"`
	Password string   `json:"password" validate:"required"`
	Remark   string   `json:"remark,omitempty"`
	Status   int      `json:"status" validate:"required"`
	IsAdmin  int      `json:"isAdmin" validate:"required"`
	RoleIDs  []string `json:"role_ids" validate:"required"`
}

type UserUpdateArgs struct {
	ID       string   `json:"id" validate:"required"`
	Username string   `json:"username,omitempty"`
	Remark   string   `json:"remark,omitempty"`
	Status   int      `json:"status,omitempty"`
	IsAdmin  int      `json:"isAdmin,omitempty"`
	RoleIDs  []string `json:"role_ids,omitempty"`
}

type UserDeleteArgs struct {
	IDs []string `json:"ids,omitempty"`
}

type RoleQueryArgs struct {
	ID     string   `json:"id,omitempty"`
	IDs    []string `json:"ids,omitempty"`
	Name   string   `json:"name,omitempty"`
	Names  []string `json:"names,omitempty"`
	Status int      `json:"status,omitempty"`
}

type RoleQueryReply struct {
	Data  []Role `json:"data"`
	Total int    `json:"total"`
}

type RoleCreateArgs struct {
	Name        string   `json:"name" validate:"required"`
	Remark      string   `json:"remark,omitempty"`
	Status      int      `json:"status" validate:"required"`
	ResourceIDs []string `json:"resource_ids,omitempty"`
}

type RoleUpdateArgs struct {
	ID          string   `json:"id" validate:"required"`
	Name        string   `json:"name,omitempty"`
	Remark      string   `json:"remark,omitempty"`
	Status      int      `json:"status,omitempty"`
	ResourceIDs []string `json:"resource_ids,omitempty"`
}

type RoleDeleteArgs struct {
	IDs []string `json:"ids,omitempty"`
}

type ResourceQueryArgs struct {
	ID string `json:"id,omitempty"`
}

type ResourceQueryReply struct {
	Data  []Resource `json:"data"`
	Total int        `json:"total"`
}

File: web/src/interface/account.ts
/**
 * @Author     : Amu
 * @Date       : 2024/09/14 10:33:31
 * @Description:
 */

export interface User {
    id: string
    username: string
    remark: string
    is_admin: number
    status: number
    created_at: string
    roles: Role[]
}

export interface UserCreateArgs {
    username: string
    password: string
    remark: string
    status: number
    role_ids: string[]
}

export interface UserUpdateArgs {
    id: string
    username: string
    remark: string
    status: number
    role_ids: string[]
}

export interface UserDeleteArgs {
    ids: string[]
}

export interface UserQueryResult {
    data: User[]
    total: number
    page: number
    size: number
}

export interface Role {
    id: string
    name: string
    remark: string
    status: number
    created_at: string
    resources: Resource[]
}

export interface RoleCreateArgs {
    name: string
    remark: string
    status: number
    resource_ids: string[]
}

export interface RoleUpdateArgs {
    id: string
    name: string
    remark: string
    status: number
    resource_ids: string[]
}

export interface RoleDeleteArgs {
    ids: string[]
}

export interface RoleQueryResult {
    data: Role[]
    total: number
    page: number
    size: number
}

export interface Resource {
    id: string
    name: string
    path: string
    method: string
    status: number
}

export interface ResourceQueryResult {
    data: Resource[]
    total: number
    page: number
    size: number
}
